conbench/entities/commit.py

```diff
@@ -157,12 +160,32 @@ def repository_to_url(repository):
 
 
 def get_github_commit(repository: str, pr_number: str, branch: str, sha: str) -> dict:
+    """
+    This function interacts with the GitHub HTTP API. Exceptions related to API
+    interaction errors are not handled here (and should be expected and handled
+    in the caller). Expected error sources are among the following :
+
+    - transient issues on DNS or TCP level
+    - transient HTTP errors such as 5xx
+    - transient but permanent-ish HTTP errors such as rate limiting
+    - HTTP authentication/authorization errors
+    - permanent error responses
+
+    If this is free of bugs then all expected exceptions should derive from
+    requests.exceptions.RequestException.
+
+    """
     if not repository or not sha:
         return {}
 
     github = GitHub()
     name = repository_to_name(repository)
+
+    # `github.get_commit()` below may raise an exception if the GitHub
+    # GitHub HTTP API failed, e.g. with a 4xx rate limiting response.
     commit = github.get_commit(name, sha)
+
+    # Note(JP): I think with current err handling this cannot happen anymore.
```

Comment:
Agreed! I think you can delete this line

Reply:
Cool. Thanks for double-checking. Added a commit!